back/engines/commercial/report_builder/app/services/report_builder/permissions_service.rb

```diff
@@ -24,7 +24,15 @@ def report_has_unauthorized_data?(report, current_user)
       next unless ReportBuilder::QueryRepository::GRAPH_RESOLVED_NAMES_CLASSES.key?(resolved_name)
 
       props = node_obj['props']
-      if props['projectId'].blank? || project_ids.exclude?(props['projectId'])
+      phase_id = props['phaseId']
+
+      if phase_id.present?
```

Comment:
You're right, but unfortunately, it creates another issue. Now we make a query to the DB for every graph. It can be fine for one report, but if (when) we have the reports page for moderators, it'll be much worse.

Reply:
Also, we have this check https://github.com/CitizenLabDotCo/citizenlab/blob/5e88b63c451db31e9bbb6e302cf0f14adcd3c4ca/back/engines/commercial/report_builder/app/policies/report_builder/live_graph_data_unit_policy.rb#L6-L8, so it will still not be possible to actually get unauthorized data.

Reply:
If it's only on the reports page for moderators in the admin view, and we have already filtered out reports that the moderator is owner of, I don't think they will have so many reports that it will become a huge problem... or do you think it will?

Reply:
In any case it's also not really a big risk to remove it, so if you think the performance consequences are serious we can also remove it

Reply:
Ok, maybe it's not significant enough to spend time on it taking into account our time constraints.